quarkus/tests/integration/src/test/java/org/keycloak/it/storage/database/ExternalInfinispanTest.java

```diff
@@ -28,10 +33,16 @@
 
 @DistributionTest(keepAlive = true)
 @WithExternalInfinispan
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class ExternalInfinispanTest {
 
     @Test
-    @Launch({ "start-dev", "--features=multi-site", "--cache=ispn", "--cache-config-file=../../../test-classes/ExternalInfinispan/kcb-infinispan-cache-remote-store-config.xml", "-Djboss.site.name=ISPN" })
+    @Launch({ "start-dev", "--features=multi-site", "--cache=ispn", "--cache-config-file=../../../test-classes/ExternalInfinispan/kcb-infinispan-cache-remote-store-config.xml", "--spi-connections-infinispan-quarkus-site-name=ISPN" })
+    // Needs to be executed as last; removing caches in this messes up with cache manager clean up after the test causing the following exception upon next start (possible ISPN bug?):
+    // org.infinispan.client.hotrod.exceptions.HotRodClientException:: ISPN004091: MBean registration failed
+    // ...
+    // Caused by: javax.management.InstanceAlreadyExistsException: org.infinispan:type=HotRodClient,name=Default,cache=sessions
```

Comment:
@keycloak/store Can you please review as well? I'm not sure particularly about this if it's a bug, or expected behaviour.

Reply:
Hi @vmuzikar - I've added something which I think is a workaround. 

I've raised an upstream issue. This should be good to be merged now.